scripts/ssl-manager.js

```diff
@@ -468,6 +472,89 @@ function SSLManager(config) {
         return me.getFileUrl("scripts/" + scriptName);
     };
 
+    me.initAddOnExtIp = function initAddOnExtIp(withExtIp) {
+        config.withExtIp = !!(withExtIp == "true");
+
+        return { result: 0 };
+    };
+
+    me.initBindedDomains = function() {
+        var domains = [],
+            domain,
+            resp;
+
+        if (config.bindedDomains) return config.bindedDomains;
+
+        resp = jelastic.env.binder.GetExtDomains(config.envName, session);
+        if (resp.result != 0) return resp;
+
+        for (var i = 0, n = resp.extDomains.length; i < n; i++) {
+            domain = resp.extDomains[i];
+            domains.push(domain.domain);
+        }
+
+        config.bindedDomains = domains.join(",");
+
+        return { result: 0 };
+    };
+
+    me.bindExtDomains = function bindExtDomains() {
+        var customDomains = config.customDomains,
+            bindedDomains = config.bindedDomains,
+            readyToGenerate = [],
+            busyDomains = [],
+            freeDomains = [],
+            domain,
+            resp;
+
+        if (customDomains) {
+            customDomains = me.parseDomains(customDomains);
+        }
+
+        for (var i = 0, n = customDomains.length; i < n; i++) {
+            domain = customDomains[i];
+
+            if (bindedDomains.indexOf(domain) != -1) {
+                readyToGenerate.push(domain);
+                continue;
+            }
+
+            if (me.isBusyExtDomain(domain)) {
+                busyDomains.push(domain)
+            } else {
+                readyToGenerate.push(domain);
+                freeDomains.push(domain);
+            }
+        }
+
+        me.setSkippedDomains(busyDomains.join(" "));
+        me.setCustomDomains(readyToGenerate.join(" "));
+
+        if (freeDomains.length) {
+            return jelastic.env.binder.BindExtDomains({
+                envName: config.envName,
+                session: session,
+                extDomains: freeDomains.join(";")
+            });
+        }
+
+        return { result: 0 };
+    };
+
+    me.isBusyExtDomain = function (domain) {
+        var BUSY_RESULT = 2330,
```

Comment:
hardcode.
Add ```import com.hivext.api.environment.response.EnvironmentResponse```
and replace with EnvironmentResponse.ENVIRONMENT_EXT_DOMAIN_IS_BUSY